dev/src/collection-group.ts

```diff
@@ -168,18 +172,23 @@ export class CollectionGroup<T = firestore.DocumentData>
    *   post.someNonExistentProperty; // TS error
    * }
    *
-   * @param {FirestoreDataConverter} converter Converts objects to and from
-   * Firestore.
+   * @param {FirestoreDataConverter | null} converter Converts objects to and
+   * from Firestore. Passing in `null` applies the default `DocumentData` typed
+   * converter.
    * @return {CollectionGroup} A `CollectionGroup<U>` that uses the provided
    * converter.
    */
+  withConverter(converter: null): CollectionGroup<firestore.DocumentData>;
   withConverter<U>(
     converter: firestore.FirestoreDataConverter<U>
+  ): CollectionGroup<U>;
+  withConverter<U>(
+    converter: firestore.FirestoreDataConverter<U> | null
   ): CollectionGroup<U> {
     return new CollectionGroup<U>(
       this.firestore,
       this._queryOptions.collectionId,
-      converter
+      converter !== null ? converter : defaultConverter()
```

Comment:
Since this is NodeJS and we don't care as much about code size, you could also use `converter ?? defaultConverter`.